File: cpp/alerts/peer_alert.cpp
#include "./peer_alert.h"

#include <memory>

namespace ltrs {

std::array<unsigned char, 20> peer_alert_get_pid(lt::peer_alert* alert) {
    std::array<unsigned char, 20> copied;
    std::memcpy(copied.data(), alert->pid.data(), copied.size());
    return copied;
}

} // namespace ltrs

File: src/ffi/alerts/peer_alert.rs
#[cxx::bridge(namespace = "ltrs")]
pub(crate) mod ffi {
    #[namespace = "libtorrent"]
    extern "C++" {
        type peer_alert = crate::ffi::ffi::peer_alert;
    }
    unsafe extern "C++" {
        include!("cpp/alerts/peer_alert.h");
        unsafe fn peer_alert_get_pid(alert: *mut peer_alert) -> [u8; 20];
    }
}
